torch/nn/functional.py

```diff
@@ -591,7 +591,19 @@ def adaptive_avg_pool3d(input, output_size):
 
 
 # Activation functions
-def dropout(input, p=0.5, training=False, inplace=False):
+def dropout(input, p=0.5, training=True, inplace=False):
```

Comment:
please do not change the default arg.

Reply:
I agree that it is kind of goofy. If you don't pass a `train` parameter, functional dropout is always a no-op and there's no point in calling it. (I don't know what the BC breaking implications of this are though.)

Reply:
That's true, I will revert the default arg and make a note at the API doc. Still wonder when will `correcting things` have higher priority than `not BC breaking`..

Reply:
I think one could reasonably make an argument that changing the default here will not break code that was already "broken" (dropout silently ignored.)

Reply:
yeah, I agree. Although it might still break someone's code if they understand the default behavior.. I can go with either way. Thoughts? cc @SsnL

Reply:
I'm fine with the change. I personally welcome it as a user as I discovered this difference in an unpleasant way. I don't think that there is much code assuming F.dropout has default `training=False` (because it would be noop w/o specification). But I'm kinda sad we can't give a warning somehow.

Reply:
you mean I should put a warning there after changing the default arg?

Reply:
alright let's just change it.

you can't make a warning because it is impossible to differentiate which users want from the calling args.